alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/groups/GroupTriggerManager.java

```diff
@@ -0,0 +1,597 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.alerter.groups;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.Resource;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.inject.Inject;
+
+import org.hawkular.alerts.api.json.GroupMemberInfo;
+import org.hawkular.alerts.api.model.condition.CompareCondition;
+import org.hawkular.alerts.api.model.condition.Condition;
+import org.hawkular.alerts.api.model.trigger.Trigger;
+import org.hawkular.alerts.api.services.DefinitionsEvent;
+import org.hawkular.alerts.api.services.DefinitionsListener;
+import org.hawkular.alerts.api.services.DefinitionsService;
+import org.hawkular.metrics.core.service.MetricsService;
+import org.hawkular.metrics.model.Metric;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachemanagerlistener.annotation.ViewChanged;
+import org.infinispan.notifications.cachemanagerlistener.event.ViewChangedEvent;
+import org.jboss.logging.Logger;
+
+/**
+ * Another feature of the Metrics Alerter is an ability to manage a Metrics Group Trigger.  A Metrics Group Trigger
+ * is an hAlerting group trigger that has a member trigger for each metric is a set scoped by a metrics tag query.
+ * The concept is similar to data-driven triggers but because metrics does not lend itself to the required
+ * source+dataId approach used by data-driven triggers, we use this approach.
+ * <p>
+ * We interact with hAlerting to pull properly tagged group triggers.  We set up a job for each to query for
+ * metrics, and maintain member triggers for each.
+ * </p><p>
+ * Each Metrics Group Trigger has several tags to guide its handling:
+ * </p><pre>
+ *   "HawkularMetrics" : "GroupTrigger"       // Required to be processed as Metrics Group Trigger
+ *   "DataIds"         : DataId1..DataIdN     // The DataIds used in the trigger conditions, required
+ *   DataId1           : tag query            // Metrics tag query for dataId1 metrics
+ *     ...
+ *   DataIdN           : tag query            // Metrics tag query for dataIdN metrics
+ *   "SourceBy"        : TagName1..TagNameN   // *see below*
+ * </pre>
```

Comment:
I agree that more documentation is needed.  It's not written yet but this feature needs an entry in the Alerting section of the Metrics' User Guide.   I'll work on that and have it prior to the metrics release.  Tutorial lessons for both the existing external alerter support and this new feaqture would also be worthwhile.